### PR TITLE
FEAT: Fresnel extension

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
## Description
This pull request introduces a new Fresnel extension for HFSS, enabling extraction of Fresnel coefficients from Floquet port simulations for periodic structures. It adds documentation, integration into the extension catalog, and comprehensive system tests for the new functionality. Additionally, several improvements and bug fixes have been made across the codebase, including enhancements to excitation handling, logging, UI theming, and design exploration.

### Fresnel Extension Integration

* Added new Fresnel extension documentation (`doc/source/User_guide/pyaedt_extensions_doc/hfss/fresnel.rst`) detailing workflows, features, validation checks, requirements, and command-line usage.
* Integrated Fresnel extension into the HFSS extension catalog

### Testing for Fresnel Extension

* Added comprehensive system tests for Fresnel extension UI, validation, and coefficient extraction, including edge cases and error handling (`tests/system/extensions/test_fresnel.py`).
* Integrated Fresnel test project fixture and extended HFSS tests to cover Fresnel port and coefficient extraction scenarios (`tests/system/general/test_hfss.py`).

### HFSS Excitation Handling Improvements

* Improved filtering of excitation names to correctly handle modes and ensure proper mapping to boundary objects (`src/ansys/aedt/core/application/analysis.py`).

### Logging and UI Enhancements

* Fixed desktop logging behavior to disable logs in non-graphical mode and enabled logs by default in extensions 
* 
* Improved UI theming by applying custom colors and fonts to SpinBox widgets 

### Design Exploration Bug Fix

* Corrected setup name mapping in design exploration to use the correct identifier

## Issue linked
**Please mention the issue number or describe the problem this pull request addresses.**

## Checklist
- [ ] I have tested my changes locally.
- [ ] I have added necessary documentation or updated existing documentation.
- [ ] I have followed the coding style guidelines of this project.
- [ ] I have added appropriate tests (unit, integration, system).
- [ ] I have reviewed my changes before submitting this pull request.
- [ ] I have linked the issue or issues that are solved by the PR if any.
- [ ] I have agreed with the Contributor License Agreement ([CLA](https://developer.ansys.com/form/cla-acceptance)).

### CODE REVIEW

**Comment by Samuelopez-ansys** [comment]
Hi @maxcapodi78 @SMoraisAnsys, could you please review this new extension?

File: src/ansys/aedt/core/application/analysis.py [hunk 1]
```diff
@@ -535,6 +535,9 @@ def design_excitations(self) -> dict[str, BoundaryObject]:
         """
         exc_names = self.excitation_names[::]
 
+        # Filter modes
+        exc_names = list(set([item.split(":")[0] for item in exc_names]))
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/application/analysis.py]:
```suggestion
        exc_names = list(set(item.split(":")[0] for item in exc_names))
```
No need for an additional list comprehension as iterating directly should be enough.

File: src/ansys/aedt/core/extensions/hfss/fresnel.py [hunk 2]
```diff
@@ -0,0 +1,1058 @@
...
+
+        self.build_advanced_tab()
+        self.build_extraction_tab()
+        self.build_settings_tab()
+
+        self.root.minsize(MIN_WIDTH, MIN_HEIGHT)
+        self.root.maxsize(MAX_WIDTH, MAX_HEIGHT)
+        self.root.geometry(f"{WIDTH}x{HEIGHT}")
+
+    def on_fresnel_type_changed(self):  # pragma: no cover
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/extensions/hfss/fresnel.py]:
```suggestion
    def on_fresnel_type_changed(self):
```

File: src/ansys/aedt/core/extensions/hfss/fresnel.py [hunk 3]
```diff
@@ -0,0 +1,1058 @@
...
+        self.active_setup = self.setups[self.setup_sweep_names[0].split(" : ")[0]]
+
+        # Validate button
+        self._widgets["validate_button"] = ttk.Button(
+            self._widgets["extraction_tab"],
+            text="Validate",
+            width=40,
+            command=lambda: self.validate(),
+            style="PyAEDT.TButton",
+        )  # nosec
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/extensions/hfss/fresnel.py]: Why are you adding `# nosec` here ? Same comment for later calls.

File: src/ansys/aedt/core/extensions/hfss/fresnel.py [hunk 4]
```diff
@@ -0,0 +1,1058 @@
...
+
+        # Floquet and modes
+
+        # Check number of ports and each port should have 2 modes
+        self.floquet_ports = self.aedt_application.get_fresnel_floquet_ports()
+        if self.floquet_ports is None:
+            self._widgets["floquet_ports_label_extraction"]["text"] = "❌"
+            return False
+        self._widgets["floquet_ports_label_extraction"]["text"] = (
+            f"{len(self.floquet_ports)} Floquet port defined" + " ✅"
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/extensions/hfss/fresnel.py]: I'm not sure that adding this kind of characters (❌ ✅) won't back fire at some point :/

File: src/ansys/aedt/core/extensions/hfss/fresnel.py [hunk 5]
```diff
@@ -0,0 +1,1058 @@
...
+        self.aedt_application.analyze_setup(cores=cores, num_variations_to_distribute=tasks, name=active_parametric)
+
+        self.aedt_application.save_project()
+
+        is_valid = self.validate(self.active_setup_sweep)
+
+        if is_valid:
+            self.get_coefficients()
+
+    def get_coefficients(self):  # pragma: no cover
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/extensions/hfss/fresnel.py]:
```suggestion
    def get_coefficients(self):
```
Remove all no cover please. Since we can't test the file yet (> 2026R1), we should just remove it from the coverage.

File: src/ansys/aedt/core/extensions/misc.py [hunk 6]
```diff
@@ -47,9 +47,11 @@
 import ansys.aedt.core.extensions
 from ansys.aedt.core.generic.design_types import get_pyaedt_app
 from ansys.aedt.core.generic.general_methods import active_sessions
+from ansys.aedt.core.generic.settings import settings
 from ansys.aedt.core.internal.aedt_versions import aedt_versions
 from ansys.aedt.core.internal.errors import AEDTRuntimeError
 
+settings.enable_desktop_logs = True
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/extensions/misc.py]: This will change the behavior of all extensions. Are you sure about that change ?

File: src/ansys/aedt/core/desktop.py [hunk 7]
```diff
@@ -691,7 +691,8 @@ def __init__(
 
         # Setup logging.
         self.__set_logger_file()
-        settings.enable_desktop_logs = not self.non_graphical and self.aedt_version_id < "2024.2"
+        if self.non_graphical:
+            settings.enable_desktop_logs = False
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/desktop.py]: Isn't `settings.enable_desktop_logs = not self.non_graphical` then ? With that change we'll be missing the case where non_graphical is false, unless enable_desktop_logs is already handled somewhere else but this would make things more complicated to follow.

File: src/ansys/aedt/core/hfss.py [hunk 8]
```diff
@@ -7704,3 +7706,607 @@ def delete_table(self, name: str | list | None = None) -> bool:
         # UI is not updated, and it needs to save the project
         self.save_project()
         return True
+
+    @pyaedt_function_handler()
+    def create_fresnel_variables(self, setup_sweep: str) -> None:
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/hfss.py:7711]: Return value doesn't align with docstring (nothing defined) and return value.

File: src/ansys/aedt/core/hfss.py [hunk 9]
```diff
@@ -7704,3 +7706,607 @@ def delete_table(self, name: str | list | None = None) -> bool:
...
+        bottom ports based on their bounding box Z-coordinate. The function
+        supports only two Floquet ports. If more than two ports are defined or
+        the mode conditions are not met, appropriate errors are logged.
+
+        Returns
+        -------
+        list of str
+            A list containing the names of the top and bottom Floquet ports,
+            identified by their vertical position (Z-axis). If validation fails,
+            returns ``None``.
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/hfss.py]: This won't return None if I'm not mitaken

**Review by SMoraisAnsys — COMMENTED** [review]
I can't test the behavior as I don't have the AEDT version but I left some comments.
Would it be possible to improve a bit the extension UI layout ? :)

File: src/ansys/aedt/core/application/analysis.py
Comment (eblanco-ansys) [inline src/ansys/aedt/core/application/analysis.py]: If `variable` and `expression` are optional you need to add the `| None = None`

Comment (eblanco-ansys) [inline src/ansys/aedt/core/application/analysis.py]:
```suggestion
        self, variable: str | None = None, expression: str | None = None, solution: str | None = None, context: list str | None = None, is_differential: bool = False
```

File: src/ansys/aedt/core/extensions/hfss/fresnel.py [hunk 10]
```diff
@@ -0,0 +1,1058 @@
...
+        self.active_setup_sweep = None
+        self.floquet_ports = None
+        self.active_parametric = None
+        self.start_frequency = None
+        self.stop_frequency = None
+        self.step_frequency = None
+        self.frequency_units = None
+
+        # Layout
+        self.root.columnconfigure(0, weight=1)
```
Comment (eblanco-ansys) [inline src/ansys/aedt/core/extensions/hfss/fresnel.py:124]: All tkinter contents should be included into `add_extension_content` function to align with the others extensions format

File: src/ansys/aedt/core/extensions/hfss/fresnel.py [hunk 11]
```diff
@@ -0,0 +1,1058 @@
...
+        self.active_setup = self.aedt_application.design_setups[simulation_setup]
+
+        self.start_frequency = float(self._widgets["start_frequency"].get("1.0", tkinter.END).strip())
+        self.stop_frequency = float(self._widgets["stop_frequency"].get("1.0", tkinter.END).strip())
+        self.step_frequency = float(self._widgets["step_frequency"].get("1.0", tkinter.END).strip())
+        self.frequency_units = self._widgets["frequency_units_combo"].get()
+
+        if self.start_frequency > self.stop_frequency:
+            self.aedt_application.logger.add_error_message("Start frequency must be less than stop frequency.")
+            self._widgets["frequency_points_label"].config(text="❌")
```
Comment (eblanco-ansys) [inline src/ansys/aedt/core/extensions/hfss/fresnel.py]: All "❌" or "✅" emojis shouldn't be included in logs or the ui to ensure consistency across all platforms

File: src/ansys/aedt/core/extensions/hfss/toolkits_catalog.toml [hunk 12]
```diff
@@ -58,6 +58,12 @@ script = "mcad_assembly.py"
 template = "run_pyaedt_toolkit_script"
 url = "https://aedt.docs.pyansys.com/version/stable/User_guide/pyaedt_extensions_doc/hfss/mcad_assembly.html"
 
+[Fresnel]
+name = "Fresnel"
+script = "fresnel.py"
+icon = "images/large/fresnel.png"
+template = "run_pyaedt_toolkit_script"
+
```
Comment (eblanco-ansys) [inline src/ansys/aedt/core/extensions/hfss/toolkits_catalog.toml:67]: Don't forget to add the url here. It's going to be something like "https://aedt.docs.pyansys.com/version/stable/User_guide/pyaedt_extensions_doc/hfss/fresnel.html"

File: src/ansys/aedt/core/extensions/misc.py (hunk 6 above)
Comment (eblanco-ansys) [reply src/ansys/aedt/core/extensions/misc.py]: x2, why this change?

File: src/ansys/aedt/core/extensions/hfss/fresnel.py (hunk 3 above)
Comment (Samuelopez-ansys) [reply src/ansys/aedt/core/extensions/hfss/fresnel.py]: Months ago Codacy complained, but now it does not anymore, I will delete it, thanks

File: src/ansys/aedt/core/extensions/misc.py (hunk 6 above)
Comment (Samuelopez-ansys) [reply src/ansys/aedt/core/extensions/misc.py]: Just a typo for testing, thanks for the catch

File: src/ansys/aedt/core/desktop.py (hunk 7 above)
Comment (Samuelopez-ansys) [reply src/ansys/aedt/core/desktop.py]: We can think about this in a new PR, I will revert it

File: src/ansys/aedt/core/application/analysis.py
Comment (Samuelopez-ansys) [reply src/ansys/aedt/core/application/analysis.py]: @eblanco-ansys They are not optionals, why do you think so?

File: tests/unit/extensions/test_fresnel.py [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+@pytest.fixture
+def mock_hfss_app_fresnel(mock_hfss_app):
+    """Fixture to create a mock AEDT application (extends HFSS mock)."""
+    with patch.object(ExtensionCommon, "desktop", new_callable=PropertyMock) as mock_desktop_property:
+        mock_desktop_instance = MagicMock()
+        mock_desktop_instance.design_list.return_value = [
+            "Design1",
+            "Design2",
+        ]
+        mock_desktop_instance.aedt_version_id = "2026.2"
```
Comment (SMoraisAnsys) [inline tests/unit/extensions/test_fresnel.py]:
```suggestion
        mock_desktop_instance.aedt_version_id = "2026.1"
```
Should this be udpated ?

Comment (Samuelopez-ansys) [reply tests/unit/extensions/test_fresnel.py]: There are one functionality, that does not work yet, until 2027R1

File: tests/unit/extensions/test_fresnel.py [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+@pytest.fixture
+def mock_hfss_app_fresnel(mock_hfss_app):
+    """Fixture to create a mock AEDT application (extends HFSS mock)."""
+    with patch.object(ExtensionCommon, "desktop", new_callable=PropertyMock) as mock_desktop_property:
+        mock_desktop_instance = MagicMock()
+        mock_desktop_instance.design_list.return_value = [
+            "Design1",
+            "Design2",
+        ]
+        mock_desktop_instance.aedt_version_id = "2026.1"
```
Comment (Samuelopez-ansys) [inline tests/unit/extensions/test_fresnel.py:46]:
```suggestion
        mock_desktop_instance.aedt_version_id = "2027.1"
```

File: tests/unit/extensions/test_fresnel.py [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+@pytest.fixture
+def mock_hfss_app_fresnel(mock_hfss_app):
+    """Fixture to create a mock AEDT application (extends HFSS mock)."""
+    with patch.object(ExtensionCommon, "desktop", new_callable=PropertyMock) as mock_desktop_property:
+        mock_desktop_instance = MagicMock()
+        mock_desktop_instance.design_list.return_value = [
+            "Design1",
+            "Design2",
+        ]
+        mock_desktop_instance.aedt_version_id = "2027.1"
```
Comment (Samuelopez-ansys) [inline tests/unit/extensions/test_fresnel.py]:
```suggestion
        mock_desktop_instance.aedt_version_id = "2026.1"
```
